### DISCUSSION TITLE
Copy and Paste layers

### DISCUSSION THREAD

**Discussion by hardusviviers** [keiwando/toonsquid#129] [Ideas]
Hello!

I've recently switched over to toonsquid from rough animator and one of the main things I'm missing and used the most is easy copy & pasting. 

My main issue is the splitting of layers and having to merge them again.
I simply want to copy from one frame and paste it into another.
Here is a demonstration I made, to help explain.
https://user-images.githubusercontent.com/90702569/211832485-216a89f4-b21f-41a6-b65b-7fb5244994fe.mp4

Also while browsing youtube I found another app, called Callipeg.
In this video they show a feature where they made it really easy to copy and paste, it's from 0:55 - 1:05
[https://youtu.be/XZ4qpioqb34?t=55](https://youtu.be/XZ4qpioqb34?t=55)
I'd imagine selecting a part of a drawing with the lasso tool, copying it, and pasting it into a new frame without making new layers and having to merge everything again or delete layers.

I saw the one discussion about making copy/paste/cut separate buttons and that would be helpful, as long as there is an option available to copy/paste it on the current layer you are on.

This app is a game-changer, 
thank you :)

**Comment by DinoAwakening** [comment]
I moved from RoughAnimator aswell! Also missed that easy copy-paste tho… So I made it one of my concepts, and Keiwan already saw it. I hope he add this one specifically on the next update! :D  

https://youtu.be/i2ZlolZOH98

**Reply by hardusviviers** [reply]
the copy feature looks really nice! that's exactly what I need, just a simple copy and paste in the same layer you are currently in >:D

**Reply by DinoAwakening** [reply]
Love to hear you liked my concept! Now on, what's left is hoping Keiwan add it (or some of them) in the next update :D
